### ISSUE TITLE
TCPDF Deprecation errors on PHP 8.0

### ISSUE THREAD

**Issue by dpfender44** [phpmyadmin/phpmyadmin#16268]
Testing phpmyadmin 5.0 and 5.1 yielded many PHP errors with PHP 8.0.0 Alpha 2 on the Export tab. All other tabs were good.
```
Deprecation Notice in .\vendor\tecnickcom\tcpdf\include\tcpdf_fonts.php#1998
 Required parameter $currentfont follows optional parameter $isunicode
Deprecation Notice in .\vendor\tecnickcom\tcpdf\include\tcpdf_fonts.php#2024
 Required parameter $currentfont follows optional parameter $isunicode
Deprecation Notice in .\vendor\tecnickcom\tcpdf\include\tcpdf_fonts.php#2040
 Required parameter $currentfont follows optional parameter $setbom
Deprecation Notice in .\vendor\tecnickcom\tcpdf\include\tcpdf_fonts.php#2060
 Required parameter $currentfont follows optional parameter $setbom
Deprecation Notice in .\vendor\tecnickcom\tcpdf\include\tcpdf_fonts.php#2077
 Required parameter $currentfont follows optional parameter $str
Deprecation Notice in .\vendor\tecnickcom\tcpdf\include\tcpdf_fonts.php#2093
 Required parameter $currentfont follows optional parameter $str
Deprecation Notice in .\vendor\tecnickcom\tcpdf\include\tcpdf_static.php#1139
 Required parameter $tagvspaces follows optional parameter $default_css
Deprecation Notice in .\vendor\tecnickcom\tcpdf\include\tcpdf_static.php#2510
 Required parameter $k follows optional parameter $points
```

**Comment by williamdes** [comment]
Thank you @dpfender44  I forwarded it upstream a while ago
See: https://github.com/tecnickcom/TCPDF/issues/222

**Comment by williamdes** [comment]
https://github.com/tecnickcom/TCPDF/pull/293 was merged but not tag was pushed for now

**Comment by williamdes** [comment]
https://github.com/tecnickcom/TCPDF/pull/265 was closed.

Thank you @brainfoolong for the effort you had made.
We welcome contributions here if you want :)

**Comment by brainfoolong** [comment]
@williamdes Thank you, i understand :) Unfortunetely i do not use Phpmyadmin, so i can't invest time here. Have a nice day.

**Comment by wizwych** [comment]
I have managed to make a temporary fix for this my issue was I couldn't use the EXPORT function from Phpmyadmin without getting the errors.. so anything that had EXAMPLE....

1998 public static function UTF8StringToArray($str, $isunicode=true, &$currentfont)

I had to change to 

1998 public static function UTF8StringToArray($str, $isunicode, &$currentfont)

as with the following lines from vendor\tecnickcom\tcpdf\include\tcpdf_fonts.php

2024 public static function UTF8ToLatin1($str, $isunicode, &$currentfont)
2040 public static function UTF8ToUTF16BE($str, $setbom, $isunicode, &$currentfont)
2060 public static function utf8StrRev($str, $setbom, $forcertl, $isunicode, &$currentfont)
2077 public static function utf8StrArrRev($arr, $str, $setbom, $forcertl, $isunicode, &$currentfont)
2093 public static function utf8Bidi($ta, $str, $forcertl, $isunicode, &$currentfont)

also file vendor\tecnickcom\tcpdf\include\tcpdf_static.php

lines  to be changed

1139 public static function fixHTMLCode($html, $default_css='', $tagvs='', $tidy_options='', &$tagvspaces)
2510 public static function setPageBoxes($page, $type, $llx, $lly, $urx, $ury, $points=false, $k, $pagedim=array())

the edits above are just portions of the code from each line but shows what to replace aswell as keeping extra line data..

my export now works on 
phpmyadmin version = Version information: 5.0.4
PHP version 8.0...

hope this helps anytone

**Comment by ibennetch** [comment]
Thanks for the report, this will be fixed with version 5.1.0.

**Comment by IlyaIndigo** [comment]
If this cannot be fixed in 5.0, please mark this branch as incompatible with php8.

**Comment by mcdavidsh** [comment]
> I have managed to make a temporary fix for this my issue was I couldn't use the EXPORT function from Phpmyadmin without getting the errors.. so anything that had EXAMPLE....
> 
> ```
> 1998 public static function UTF8StringToArray($str, $isunicode=true, &$currentfont)
> ```
> 
> I had to change to
> 
> ```
> 1998 public static function UTF8StringToArray($str, $isunicode, &$currentfont)
> ```
> 
> as with the following lines from vendor\tecnickcom\tcpdf\include\tcpdf_fonts.php
> 
> ```
> 2024 public static function UTF8ToLatin1($str, $isunicode, &$currentfont)
> 2040 public static function UTF8ToUTF16BE($str, $setbom, $isunicode, &$currentfont)
> 2060 public static function utf8StrRev($str, $setbom, $forcertl, $isunicode, &$currentfont)
> 2077 public static function utf8StrArrRev($arr, $str, $setbom, $forcertl, $isunicode, &$currentfont)
> 2093 public static function utf8Bidi($ta, $str, $forcertl, $isunicode, &$currentfont)
> ```
> 
> also file vendor\tecnickcom\tcpdf\include\tcpdf_static.php
> 
> lines to be changed
> 
> ```
> 1139 public static function fixHTMLCode($html, $default_css, $tagvs, $tidy_options, &$tagvspaces)
> 2510 public static function setPageBoxes($page, $type, $llx, $lly, $urx, $ury, $points, $k, $pagedim=array())
> ```
> 
> the edits above are just portions of the code from each line but shows what to replace aswell as keeping extra line data..
> 
> my export now works on
> phpmyadmin version = Version information: 5.0.4
> PHP version 8.0...
> 
> hope this helps anyone


Worked also in PHP 8.0.2, phpmyadmin version = Version information: 5.0.4

**Comment by Shakil-Shahadat** [comment]
@mcdavidsh this issue has been fixed in 5.1.0.
